include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp

```diff
@@ -131,16 +132,95 @@ class affine_cell_proxy : public tuple_t
     /*!\name Constructors, destructor and assignment
      * \{
      */
-    affine_cell_proxy()= default; //!< Defaulted.
+    affine_cell_proxy() = default; //!< Defaulted.
     affine_cell_proxy(affine_cell_proxy const &) = default; //!< Defaulted.
     affine_cell_proxy(affine_cell_proxy &&) = default; //!< Defaulted.
     affine_cell_proxy & operator=(affine_cell_proxy const &) = default; //!< Defaulted.
     affine_cell_proxy & operator=(affine_cell_proxy &&) = default; //!< Defaulted.
     ~affine_cell_proxy() = default; //!< Defaulted.
 
-    // Inherit the tuple constructors and assignment.
+    // Inherit the base class's constructor to enable element-wise initialisation (direct and converting constructor).
     using tuple_t::tuple_t;
-    using tuple_t::operator=;
+
+    //!\brief Converting constructor. Initialises from another tuple type.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t &&>
+    //!\endcond
+    explicit affine_cell_proxy(other_tuple_t && other) :
+        tuple_t{std::forward<other_tuple_t>(other)}
+    {}
+
+    //!\brief Converting copy-constructor.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t &>
+    //!\endcond
+    explicit affine_cell_proxy(affine_cell_proxy<other_tuple_t> & other) :
+        tuple_t{static_cast<other_tuple_t &>(other)}
+    {}
+
+    //!\brief Converting copy-constructor.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t const &>
+    //!\endcond
+    explicit affine_cell_proxy(affine_cell_proxy<other_tuple_t> const & other) :
+        tuple_t{static_cast<other_tuple_t const &>(other)}
+    {}
+
+    //!\brief Converting move-constructor.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t>
+    //!\endcond
+    explicit affine_cell_proxy(affine_cell_proxy<other_tuple_t> && other) :
+        tuple_t{static_cast<other_tuple_t &&>(std::move(other))}
+    {}
+
+    //!\brief Converting assignment. Initialises from another tuple type.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::assignable_from<tuple_t &, other_tuple_t &&>
+    //!\endcond
+    affine_cell_proxy & operator=(other_tuple_t && other)
+    {
+        as_base() = std::forward<other_tuple_t>(other);
+        return *this;
+    }
+
+    //!\brief Converting copy-assignment.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::assignable_from<tuple_t &, other_tuple_t &>
+    //!\endcond
+    affine_cell_proxy & operator=(affine_cell_proxy<other_tuple_t> & other)
```

Comment:
Is this overload really necessary?

Shouldn't every assignment operator only be define on `const &` and `&&`? Because it needs to copy/move the data anyway.

So a tuple implementation should always provide only those two `const & / &&`?!

That means `other_tuple_t &` and `other_tuple_t const &` use the same assignment overload?

Reply:
Ok, I couldn't find a problem after removing them. Not sure, what I saw, or in which in-between state I was adding this. But many thanks for looking at this with different eyes! 👍